runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -3623,7 +3624,7 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
          // This behaviour is consistent with non-JITaaS
          ((TR_J9JITaaSServerSharedCache *) _vm->sharedCache())->setStream(stream);
 
-      if (!ramMethod)
+      if ((!ramMethod) || (entry._useAotCompilation && !enableJITaaSRemoteAOT))
```

Comment:
I do not understand the need for this code. The case with ramMethod==0 happens when the client terminates and wants to inform the server that it can delete all cached data related to this client.

Reply:
The server shouldn't process the AOT compilation request when enableJITaaSRemoteAOT is false on the server side. In this case, we need to abort the compilation request received from the client side. I thought we could piggyback on this `ramMethod==0` case to quit the compilation on the server side. Maybe not? What would be a better way to quit/ignore the compilation request in this stage on the server side?

Reply:
I would like to understand why `enableJITaaSRemoteAOT` is needed at the server side. As long as the capability exists (and currently we have this capability) the server should do whatever the client tells it to do. If the server is not supposed to perform an AOT compilation the client should not have sent the request in the first place.

Reply:
Good point. `enableJITaaSRemoteAOT` doesn't  apply to the server anymore. The following code should to be changed: 
```
eligibleForRelocatableCompile =
            (enableJITaaSRemoteAOT || !entry->isOutOfProcessCompReq()) && // disable remote JITaaS AOT until it's working
````
to 
```
eligibleForRelocatableCompile =
            (enableJITaaSRemoteAOT || entry->isOutOfProcessCompReq()) &&
````

Reply:
I would say that if we are at the server `eligibleForRelocatableCompile` should always be true because all of the preliminary checks have already been done at the client. We just need some information from the client to the server that it wants an AOT compilation.

Reply:
I see this being passed in `entry._useAotCompilation  = std::get<12>(req);` so we have the info that the client wants an AOT compilation

Reply:
```
  // If we are allowed to do relocatable compilations, just do it
   // Note that currently we don't support out-of-process relocatable compilations
   if (canDoRelocatableCompile)
      {
      _vm = TR_J9VMBase::get(_jitConfig, vmThread, TR_J9VMBase::AOT_VM);
      entry->_useAotCompilation = true;

      if (_compInfo.getPersistentInfo()->getJITaaSMode() == CLIENT_MODE && enableJITaaSRemoteAOT)
         {
         entry->setRemoteCompReq();
         }
      }
```
The comment needs some adjustment.
The test of `enableJITaaSRemoteAOT` could disappear and maybe in another PR replaced with a test on how expensive the compilation is expected to be. Cheap ones could be done locally while expensive ones could be done remotely.

Reply:
@dmitry-ten and I are working together to remove `TR_EnableJITaaSRemoteAOT` on both the client and the server side.